tools/src/org.graalvm.tools.lsp/src/org/graalvm/tools/lsp/server/request/SignatureHelpRequestHandler.java

```
@@ -0,0 +1,85 @@
+            SourceSectionFilter inputFilter = SourceSectionFilter.ANY;
+            EvaluationResult evalResult = sourceCodeEvaluator.runToSectionAndEval(surrogate, signatureSection, eventFilter, inputFilter);
+            if (evalResult.isEvaluationDone() && !evalResult.isError()) {
+                Object result = evalResult.getResult();
+                if (result instanceof TruffleObject) {
+                    try {
+                        Object signature = ForeignAccess.send(GET_SIGNATURE, (TruffleObject) result);
+                        if (signature == null) {
+                            return new SignatureHelp();
+                        }
+                        String formattedSignature = ForeignAccess.sendInvoke(INVOKE, (TruffleObject) signature, "format").toString();
+                        List<Object> params = ObjectStructures.asList(new ObjectStructures.MessageNodes(), (TruffleObject) signature);
+                        SignatureInformation info = new SignatureInformation(formattedSignature);
+                        List<ParameterInformation> paramInfos = new ArrayList<>();
+                        for (Object param : params) {
+                            if (param instanceof TruffleObject) {
+                                Object formattedParam = ForeignAccess.sendInvoke(INVOKE, (TruffleObject) param, "format");
+                                paramInfos.add(new ParameterInformation(formattedParam.toString()));
+                            }
+                        }
+                        info.setParameters(paramInfos);
+                        Object nodeObject = nodeAtCaret.getNodeObject();
+                        Integer numberOfArguments = InteropUtils.getNumberOfArguments(nodeObject);
+
+                        return new SignatureHelp(Arrays.asList(info), 0, numberOfArguments != null ? numberOfArguments - 1 : 0);
+                    } catch (UnsupportedMessageException | UnsupportedTypeException e) {
```

never swallow exceptions